src/clj/game/cards/programs.clj

```diff
@@ -877,4 +877,48 @@
                    :counter-cost [:power 3]
                    :once :per-turn
                    :msg "gain [Click][Click]"
-                   :effect (effect (gain :click 2))}]}})
+                   :effect (effect (gain :click 2))}]}
+
+   "Wari"
+   {:events {:successful-run
+             {:interactive (req )
+              :delayed-completion true
+              :req (req (and (= target :hq)
+                             (first-successful-run-on-server? state :hq)
+                             (some #(and (ice? %)
+                                         (not (rezzed? %)))
+                                   (all-installed state :corp))))
+              :effect (effect
+                       (continue-ability
+                        {:prompt "Use Wari?"
+                         :choices ["Yes" "No"]
+                         :delayed-completion true
+                         :effect (effect 
+                                  (continue-ability
+                                   (if (= target "Yes")
+                                     {:prompt "Choose a subtype"
+                                      :choices ["Barrier" "Code Gate" "Sentry"]
+                                      :delayed-completion true
+                                      :effect (effect
+                                               (trash card)
+                                               (continue-ability
+                                                (let [chosen-subtype target]
+                                                  {:choices {:req #(and (ice? %)
+                                                                        (not (rezzed? %)))}
+                                                   :delayed-completion true
+                                                   :msg (str "name " chosen-subtype)
+                                                   :effect (req
+                                                            (when-completed (expose state side target)
+                                                              (if (and async-result
+                                                                       (has-subtype? target chosen-subtype))
+                                                                (continue-ability
```

Comment:
This final effect doesn't need a continue-ability, you can just directly code the effect functions (`move`, `system-msg`, `effect-completed`) here inside of a `do`. You _typically_ only need `continue-ability` if you need to show a new prompt,.